lib.cpp

```diff
@@ -0,0 +1,15 @@
+#include "lib.hpp"
+
+void pushChunk(unsigned char*& _chunk) {
+    int sz = 4096; 
+    _chunk = new unsigned char[sz];
+
+    for(int i=0; i<sz; i++)
+        _chunk[i] = i % 255;
+
+    chunk = _chunk;
```

Comment:
I think it should be possible that there is more then one chunk on the fly

Reply:
There're potentially two possible ways to that I guess.
1- Modifying `main.cpp` to:
  ```
  // ...
    int main() {
    std::vector<unsigned char*> chunksVec(10);
    for (auto& chnk : chunksVec) {
        pushChunk(chnk);
        // ...
       }
     return 0;
     }
```
2- Also use a vector of `unsigned char*` in `lib.hpp` and have `popChunk()` return the vector.

Reply:
Instantiate two separate queues in the `lib.cpp`. 
`push` should accept an `unsigned char*` and then push it into it's internal `emptyQueue`. 
A process in its own thread pops out one pointer of the queue, fills the data into the memory-location, and pushes the pointer to the `dataQueue` which is then eventually read (or never read) by the consumer-program.

Hope this makes sense

Reply:
```
void pushChunk(unsigned char*& chnk) {

	std::deque<unsigned char*> emptyQueue;
	std::deque<unsigned char*> dataQueue;

	// allocate memory block
	constexpr int sz = 4096;
	chnk = new unsigned char[sz];
	
	// push the chunk into the queue
	emptyQueue.push_back(chnk);

	auto fill = [](unsigned char*& _chnk) {
		for (int i = 0; i < sz; i++)
			_chnk[i] = i % 255;
		};
	
	if (!emptyQueue.empty()) {
		{
			std::jthread t(fill, std::ref(emptyQueue.front()));
		}
		dataQueue.emplace_back(std::move(emptyQueue.front()));
		emptyQueue.pop_front();
	}
}
```